src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItem.java

```diff
@@ -1,29 +1,110 @@
+import pl.tomaszdziurko.jvm_bloggers.UTMParameters;
 import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
 import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog;
 
-
-@Data
+/**
+ * {@link BlogPost} representation for email newsletter template purpose.
+ */
+@Getter
 class BlogPostForMailItem {
-
+    
+    private static final String UTM_SOURCE = "jvmbloggers";
+    private static final String UTM_MEDIUM = "newsletter";
+    
     private String title;
     private String url;
     private String authorLabel;
 
-    public BlogPostForMailItem(BlogPost blogPost) {
-        this.title = blogPost.getTitle();
-        this.url = blogPost.getUrl();
-        this.authorLabel = determineAuthorLabel(blogPost.getBlog());
+    /**
+     * Returns internal builder.
+     * 
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
     }
+    
+    /**
+     * Builder for {@link BlogPostForMailItem}
+     */
+    public static class Builder {
 
-    private String determineAuthorLabel(Blog blogger) {
-        if (blogger.getTwitter() != null) {
-            return "<a href=\"https://twitter.com/" + blogger.getTwitter().substring(1) + "\">" + blogger.getAuthor() + "</a>";
-        } else {
-            return blogger.getAuthor();
+        private final BlogPostForMailItem instance = new BlogPostForMailItem();
+
+        public Builder from(BlogPost blogPost) {
+            instance.title = blogPost.getTitle();
+            instance.url = blogPost.getUrl();
+            instance.authorLabel = determineAuthorLabel(blogPost.getBlog());
+            return this;
+        }
+        
+        public Builder withTitle(String title) {
+            instance.title = title;
+            return this;
         }
-    }
 
+        public Builder withUrl(String url) {
+            instance.url = url;
+            return this;
+        }
+        
+        public Builder withAuthorLabel(Blog blog) {
+            instance.authorLabel = determineAuthorLabel(blog);
+            return this;
+        }
+        
+        public Builder withUrlParameter(String name, String value) {
+            Preconditions.checkState(
+                instance.url != null,
+                "Url could not be null. Please set url first"
+            );
+            
+            instance.url = UriComponentsBuilder
+                .fromHttpUrl(instance.url)
+                .queryParam(name, value)
+                .build().toString();
+            
+            return this;
+        }
+        
+        public Builder withDefaultUTMParameters() {
+            withUrlParameter(UTMParameters.UTM_SOURCE_KEY, UTM_SOURCE);
+            withUrlParameter(UTMParameters.UTM_MEDIUM_KEY, UTM_MEDIUM);
+            withUrlParameter(UTMParameters.UTM_CAMPAING_KEY,
+                String.format(
+                    "%s#%s", UTM_SOURCE,
+                    new SimpleDateFormat("yyyy-mm-dd").format(new Date())
```

Comment:
`BlogSummaryMailSender` has injected `IssueNumberRetriever` and calls `issueNumberRetriever.getNextIssueNumber()` which ask database for next sequence number. I could not inject `IssueNumberRetriever` into `BlogSummaryMailGenerator` because using `issueNumberRetriever.getNextIssueNumber()` increments database sequence again (that's my guess). Please decide if it is worth refactoring this code somehow or just stay with date and `NowProvider` solution.